src/dnf-plugins/subscription-manager.py

```diff
@@ -142,14 +142,21 @@ def _warnOrGiveUsageMessage(self):
             if msg:
                 logger.info(msg)
 
+    def sack(self):
+        # If command demands resolving, it should run transaction. This cannot
+        # be run in a Plugin.config hook as currently the Plugin.config is
+        # called before Command.configure.
+        if not self.cli.demands.resolving:
+            self._package_profile()
+
     def transaction(self):
+        self._package_profile()
```

Comment:
Same as above, we should make sure we do not actually perform the _package_profile function if we already have.

Reply:
Same as above.